File: core4/leetcode/Arrays/P26/main.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (nums.empty())  return -1;
       
        int replace = 1;
       
        for(int i =1 ; i< nums.size(); i++){
            if (nums[i] != nums[i-1]){
                nums[replace] = nums[i];
                replace++;
            }
 
        }
       
        return replace;
       
    }
};

int main(){
   
    Solution object;
   
    vector<int> nums = {5,6,7,7,7,8};
   
   
    // vector can not be printed with cout
    int  k = object.removeDuplicates(nums) ;
   
    cout << "Number of unique elements:" << k << endl;
   
    cout << "New Nums"<< endl;
   
    for(int i=0; i<k;i++){
        cout << nums[i];
    }
   
   
   
   
   
   
   
   
}

File: core4/leetcode/Arrays/P26/main.py
class Solution:
    
    def RemoveDuplicate(self, nums):
        
        if not nums:
            return  0;
            
        replace = 1;
        
        for i in range(1,len(nums)):
            if nums[i-1] != nums[i]:
                nums[replace] = nums[i];
                replace+=1 
        
        return replace
    
    

if __name__ == "__main__":
    nums = [5,6,7,7,7,8]
    
    solve = Solution()
    
    R = solve.RemoveDuplicate(nums)
    
    print(f"Number of unique elements is: {R}")
    print("New array:", nums[:R])
    



